### PR TITLE
[DSL] Expose ColorUtil methods to DSL rules

### PR DESCRIPTION
**Pull request by holgerfriedrich**
* Expose xyToKelvin and kelvinToXY
* Add missing throws declaration
* Expose xyToDuv

Unfortunately, we forgot about the throws declaration in #4367. This is API breaking, you might need to add a catch statement in binding code. We did not find any issues in the add-on repo so far, it still compiles fine.

### CODE REVIEW

**Review by andrewfg — APPROVED** [review]
LGTM many thanks.

**Comment by holgerfriedrich** [comment]
Lets wait for #4401, then we can add ```xyToDuv``` as well.

**Comment by holgerfriedrich** [comment]
Done.
I will create an additional PR in docs.
@florian-h05 do you want to create a change in JS scripting as well (follow-up to openhab/openhab-js#319)?

**Comment by florian-h05** [comment]
These methods are just added to the CoreUtil class, correct?
In that case only a doc update in openhab-js is needed.

**Comment by andrewfg** [comment]
@holgerfriedrich apropos the thrown exception potentially breaking the API — two things..
1. In https://github.com/openhab/org.openhab.binding.zigbee/pull/855 I did not call the respective method, so that one will not be broken.
2. Did you check the ColorUtil JUnit tests (here in core) since they do call the respective method?

**Comment by holgerfriedrich** [comment]
@andrewfg thanks for checking, the build seems still fine. I think in tests it is not mandatory to catch all declared exceptions....

**Review by andrewfg — APPROVED** [review]
LGTM

**Comment by holgerfriedrich** [comment]
Rebased to let CI run on ubuntu-24.04

**Review by lolodomo — APPROVED** [review]

**Review by J-N-K — APPROVED** [review]
Thanks!

File: bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java [hunk 1]
```diff
@@ -1651,9 +1651,9 @@ public static double[] kelvinToXY(double kelvin) throws IndexOutOfBoundsExceptio
      * 
      * @param xy an array with the CIE colour XY values to be converted
      * @return the colour temperature in K
-     * @throws IndexOutOfBoundsException if the wrong number of arguments is provided
+     * @throws IllegalArgumentException if the wrong number of arguments is provided
      */
-    public static double xyToKelvin(double[] xy) {
+    public static double xyToKelvin(double[] xy) throws IllegalArgumentException {
```
Comment (J-N-K) [inline bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java:1656]: Since an IAE is a runtime exception this should not break code.

**Pull request merged**